service/src/main/java/org/mskcc/cmo/metadb/service/impl/RequestReplyHandlingServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+package org.mskcc.cmo.metadb.service.impl;
+
+import java.util.List;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.Phaser;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.mskcc.cmo.messaging.Gateway;
+import org.mskcc.cmo.messaging.MessageConsumer;
+import org.mskcc.cmo.metadb.model.SampleMetadata;
+import org.mskcc.cmo.metadb.service.RequestReplyHandlingService;
+import org.mskcc.cmo.metadb.service.SampleService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.nats.client.Message;
+
+@Component
+public class RequestReplyHandlingServiceImpl implements RequestReplyHandlingService {
+    
+    @Value("${request_reply.patient_samples_topic}")
+    private String REQUEST_REPLY_TOPIC;
```

Comment:
Suggest this be called something more specific to the actual topic in the event a new request-reply topic is introduced in the future.